feature/platform/transceiver/tests/zr_input_output_power_test/zr_input_output_power_test.go

```diff
@@ -0,0 +1,411 @@
+	t.Logf("Optical channel = %v , Instant Input Power = %v", transceiverOpticalChannelName, transceiverOpticalChanelInputPower.GetInstant())
+	t.Logf("Optical channel = %v , Average Input Power = %v", transceiverOpticalChannelName, transceiverOpticalChanelInputPower.GetAvg())
+	t.Logf("Optical channel = %v , Min Input Power = %v", transceiverOpticalChannelName, transceiverOpticalChanelInputPower.GetMin())
+	t.Logf("Optical channel = %v , Max Input Power = %v", transceiverOpticalChannelName, transceiverOpticalChanelInputPower.GetMax())
+
+	if transceiverInputPowerInstantType := reflect.TypeOf(transceiverOpticalChanelInputPower.GetInstant()).Kind(); transceiverInputPowerInstantType != reflect.Float64 {
+		t.Fatalf("[Error]: Expected Optical Channel %v  Instant InputPower  data type = %v  , Got =%v", transceiverOpticalChannelName, reflect.Float64, transceiverInputPowerInstantType)
+	}
+	if transceiverInputPowerAvgType := reflect.TypeOf(transceiverOpticalChanelInputPower.GetAvg()).Kind(); transceiverInputPowerAvgType != reflect.Float64 {
+		t.Fatalf("[Error]: Expected Optical Channel %v  Avg InputPower data type = %v  , Got =%v", transceiverOpticalChannelName, reflect.Float64, transceiverInputPowerAvgType)
+	}
+	if transceiverInputPowerMinType := reflect.TypeOf(transceiverOpticalChanelInputPower.GetMin()).Kind(); transceiverInputPowerMinType != reflect.Float64 {
+		t.Fatalf("[Error]: Expected Optical Channel %v  Min InputPower data type = %v  , Got =%v", transceiverOpticalChannelName, reflect.Float64, transceiverInputPowerMinType)
+	}
+	if transceiverInputPowerMaxType := reflect.TypeOf(transceiverOpticalChanelInputPower.GetMax()).Kind(); transceiverInputPowerMaxType != reflect.Float64 {
+		t.Fatalf("[Error]: Expected Optical Channel %v Max InputPower data type = %v  , Got =%v", transceiverOpticalChannelName, reflect.Float64, transceiverInputPowerMaxType)
+	}
+
+	if isInterfaceDisabled {
+		if transceiverOpticalChanelInputPower.GetInstant() > 0 {
+			t.Fatalf("[Error]: Expected Optical Channel %v  Instant InputPower   = %v  ,Got =%v", transceiverOpticalChannelName, 0, transceiverOpticalChanelInputPower.GetInstant())
+		}
+		if transceiverOpticalChanelInputPower.GetAvg() > 0 {
+			t.Fatalf("[Error]: Expected Optical Channel %v  Avg InputPower   = %v  ,Got =%v", transceiverOpticalChannelName, 0, transceiverOpticalChanelInputPower.GetAvg())
+		}
+		if transceiverOpticalChanelInputPower.GetMin() > 0 {
+			t.Fatalf("[Error]: Expected Optical Channel %v  Min InputPower   = %v  ,Got =%v", transceiverOpticalChannelName, 0, transceiverOpticalChanelInputPower.GetMin())
+		}
+		if transceiverOpticalChanelInputPower.GetMax() > 0 {
+			t.Fatalf("[Error]: Expected Optical Channel %v  Max InputPower   = %v  ,Got =%v", transceiverOpticalChannelName, 0, transceiverOpticalChanelInputPower.GetMax())
+		}
+
+	} else if transceiverOpticalChanelInputPower.GetMin() < rxSignalPowerLowerBound || transceiverOpticalChanelInputPower.GetMax() > rxSignalPowerUpperBound {
```

Comment:
can you also check in ENABLED case avg and instant are < min and > max.

Reply:
Added in the latest commit.